### PR TITLE
Remove the deprecated 'doc-comments-val' option

### PR DESCRIPTION
**Pull request by gpetiot**

### CODE REVIEW

File: lib/Conf.ml [hunk 1]
```diff
@@ -501,74 +500,9 @@ module Formatting = struct
         , "$(b,before) puts comments before the corresponding code." ) ]
     in
     C.choice ~names ~all ~doc ~section
-      ~removed_values:
-        [ C.removed_value ~name:"after" ~version:"0.14.2"
-            ~msg:
-              "This value has been renamed `after-when-possible` to take \
-               into account the technical limitations of ocamlformat, the \
-               behavior is unchanged." ]
```
Comment (Julow) [inline lib/Conf.ml:509]: Why remove this ?

**Review by Julow — COMMENTED** [review]
We should probably keep a list of removed options and a short message to help users migrate. Not very important for this option but it may be useful for https://github.com/ocaml-ppx/ocamlformat/pull/1462.
I started to work on this a while ago: https://github.com/ocaml-ppx/ocamlformat/compare/master...Julow:removed-options

File: lib/Conf.ml (hunk 1 above)
Comment (gpetiot) [reply lib/Conf.ml:509]: oh right, reverting this part!

**Comment by emillon** [comment]
Yes, I agree a quick CLI test for these would be useful.

**Comment by emillon** [comment]
Also it seems a bit early to remove this, I suggest we wait until 0.17.0 for this one.

**Comment by gpetiot** [comment]
Since this option was already effectively removed from the formatting, and only listed in the list of options from the manpage, I don't think we should hold this PR until 0.17.0, it is now listed as a removed option.

**Pull request merged**